DQMOffline/Trigger/plugins/RazorMonitor.cc

```diff
@@ -0,0 +1,371 @@
+
+  MRVsRsq_ME_.denominator -> Fill(MR, Rsq);
+
+  // applying selection for numerator
+  if (num_genTriggerEventFlag_->on() && ! num_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  // filling histograms (numerator)
+  if (Rsq>=rsqCut_) {
+    MR_ME_.numerator -> Fill(MR);
+    MRVsLS_.numerator -> Fill(ls, MR);
+  }
+
+  if (MR>=mrCut_) {
+    Rsq_ME_.numerator -> Fill(Rsq);
+    RsqVsLS_.numerator -> Fill(ls, Rsq);
+  }
+
+  dPhiR_ME_.numerator -> Fill(dPhiR);
+
+  MRVsRsq_ME_.numerator -> Fill(MR, Rsq);
+
+}
+
+void RazorMonitor::fillHistoLSPSetDescription(edm::ParameterSetDescription & pset)
+{
+  pset.add<int32_t>   ( "nbins", 2500);
+}
+
+void RazorMonitor::fillDescriptions(edm::ConfigurationDescriptions & descriptions)
+{
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>  ( "FolderName", "HLT/SUSY/Razor" );
+
+  desc.add<edm::InputTag>( "met",      edm::InputTag("pfMet") );
+  desc.add<edm::InputTag>( "jets",     edm::InputTag("ak4PFJetsCHS") );
+  desc.add<edm::InputTag>("hemispheres",edm::InputTag("hemispheresDQM"))->setComment("hemisphere jets used to compute razor variables");
+  desc.add<std::string>("metSelection", "pt > 0");
+
+  // from 2016 offline selection
+  desc.add<std::string>("jetSelection", "pt > 80");
+  desc.add<uint>("njets",      2);
+  desc.add<double>("mrCut",    300);
+  desc.add<double>("rsqCut",   0.15);
+
+  edm::ParameterSetDescription genericTriggerEventPSet;
+  genericTriggerEventPSet.add<bool>("andOr");
+  genericTriggerEventPSet.add<edm::InputTag>("dcsInputTag", edm::InputTag("scalersRawToDigi") );
+  genericTriggerEventPSet.add<std::vector<int> >("dcsPartitions",{});
+  genericTriggerEventPSet.add<bool>("andOrDcs", false);
+  genericTriggerEventPSet.add<bool>("errorReplyDcs", true);
+  genericTriggerEventPSet.add<std::string>("dbLabel","");
+  genericTriggerEventPSet.add<bool>("andOrHlt", true);
+  genericTriggerEventPSet.add<edm::InputTag>("hltInputTag", edm::InputTag("TriggerResults::HLT") );
+  genericTriggerEventPSet.add<std::vector<std::string> >("hltPaths",{});
+  genericTriggerEventPSet.add<std::string>("hltDBKey","");
+  genericTriggerEventPSet.add<bool>("errorReplyHlt",false);
+  genericTriggerEventPSet.add<unsigned int>("verbosityLevel",1);
+
+  desc.add<edm::ParameterSetDescription>("numGenericTriggerEventPSet", genericTriggerEventPSet);
+  desc.add<edm::ParameterSetDescription>("denGenericTriggerEventPSet", genericTriggerEventPSet);
+
+  //binning from 2016 offline selection
+  edm::ParameterSetDescription histoPSet;
+  std::vector<double> mrbins = {0., 100., 200., 300., 400., 500., 575., 650., 750., 900., 1200., 1600., 2500., 4000.};
+  histoPSet.add<std::vector<double> >("mrBins", mrbins);
+
+  std::vector<double> rsqbins = {0., 0.05, 0.1, 0.15, 0.2, 0.25, 0.30, 0.41, 0.52, 0.64, 0.8, 1.5};
+  histoPSet.add<std::vector<double> >("rsqBins", rsqbins);
+
+  std::vector<double> dphirbins = {0., 0.5, 1.0, 1.5, 2.0, 2.5, 2.8, 3.0, 3.2};
```

Comment:
This is the dPhi between razor hemispheres -- It's calculated on line 220 of the same file
`  double dPhiR = abs(deltaPhi(hemispheres->at(0).Phi(), hemispheres->at(1).Phi()));`